build error when doclint checks are enabled

If the `doclint` is enabled in the pom.xml and `mvn clean install` is run the following error is returned:
```
.../eclipse/microprofile/metrics/Snapshot.java:81: error: reference not found
[ERROR]      * See {@link #percentileValue()}
```

This error can be reproduced either by running `mvn clean install` in the [microprofile project](https://github.com/eclipse/microprofile) or by enabling `doclint` in the microprofile-metrics project by enabling doclint:

To enable `doclint` in the microprofile-metrics project add the following to the [pom.xml](https://github.com/eclipse/microprofile-metrics/blob/main/pom.xml):
```
    <build>
        <pluginManagement>
            <plugins>
                <plugin>
                    <groupId>org.apache.maven.plugins</groupId>
                    <artifactId>maven-javadoc-plugin</artifactId>
                    <configuration>
                        <doclint>all</doclint>
                    </configuration>
                </plugin>
            </plugins>
        </pluginManagement>
    </build>
```

--- a/api/src/main/java/org/eclipse/microprofile/metrics/Snapshot.java
+++ b/api/src/main/java/org/eclipse/microprofile/metrics/Snapshot.java
@@ -1,6 +1,6 @@
 /*
  **********************************************************************
- * Copyright (c) 2017, 2023 Contributors to the Eclipse Foundation
+ * Copyright (c) 2017, 2024 Contributors to the Eclipse Foundation
  *               2010-2013 Coda Hale, Yammer.com
  *
  * See the NOTICES file(s) distributed with this work for additional
@@ -78,7 +78,7 @@ public abstract class Snapshot {
      * Represents a percentile and its value at the moment it was sampled from the Snapshot. Percentile values of a
      * {@link Timer} are represented in units of nanoseconds.
      *
-     * See {@link #percentileValue()}
+     * See {@link Snapshot#percentileValues()}
      */
     public static class PercentileValue {
         private final double percentile;

--- a/tck/rest/src/main/java/org/eclipse/microprofile/metrics/test/MpMetricTest.java
+++ b/tck/rest/src/main/java/org/eclipse/microprofile/metrics/test/MpMetricTest.java
@@ -1,6 +1,6 @@
 /*
  **********************************************************************
- * Copyright (c) 2017, 2022 Contributors to the Eclipse Foundation
+ * Copyright (c) 2017, 2024 Contributors to the Eclipse Foundation
  *
  * See the NOTICES file(s) distributed with this work for additional
  * information regarding copyright ownership.
@@ -49,8 +49,8 @@ import jakarta.inject.Inject;
 /**
  * Rest Test Kit
  *
- * @author Heiko W. Rupp <hrupp@redhat.com>
- * @author Don Bourne <dbourne@ca.ibm.com>
+ * @author <a href="mailto:hrupp@redhat.com">Heiko W. Rupp</a>
+ * @author <a href="mailto:dbourne@ca.ibm.com">Don Bourne</a>
  */
 @SuppressWarnings("CdiInjectionPointsInspection")
 @RunWith(Arquillian.class)

